fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/utility/handler/ChirpHandlerTest.java

```diff
@@ -48,100 +36,68 @@ public class ChirpHandlerTest {
-    messageHandler.handleMessage(messageSender, channel, createLaoMessage);
-
-    chirpChannel =
-        laoRepo
-            .getLaoView(LAO_ID)
-            .getChannel()
-            .subChannel("social")
-            .subChannel(SENDER.getEncoded());
+    // Instantiate the dependencies here such that they are reset for each test
+    LAORepository laoRepo = new LAORepository();
+    laoRepo.updateLao(LAO);
+    handler = new ChirpHandler(laoRepo, socialMediaRepo);
   }
 
   @Test
-  public void testHandleAddChirp() throws DataHandlingException, UnknownLaoException {
-    MessageGeneral message = new MessageGeneral(SENDER_KEY, ADD_CHIRP, gson);
-    messageHandler.handleMessage(messageSender, chirpChannel, message);
-
-    LaoView laoView = laoRepo.getLaoView(LAO_ID);
-    Lao updatedLao = laoView.createLaoCopy();
-
-    Optional<Chirp> chirpOpt = updatedLao.getChirp(message.getMessageId());
-    assertTrue(chirpOpt.isPresent());
-    Chirp chirp = chirpOpt.get();
-
-    assertEquals(message.getMessageId(), chirp.getId());
-    assertEquals(SENDER, chirp.getSender());
-    assertEquals(TEXT, chirp.getText());
-    assertEquals(CREATION_TIME, chirp.getTimestamp());
-    assertEquals(PARENT_ID, chirp.getParentId());
-
-    Map<MessageID, Chirp> chirps = updatedLao.getAllChirps();
-    assertEquals(1, chirps.size());
-    assertEquals(chirp, chirps.get(chirp.getId()));
+  public void testHandleAddChirp() throws UnknownLaoException {
+    HandlerContext ctx = new HandlerContext(CHIRP_ID, SENDER, CHIRP_CHANNEL, messageSender);
+
+    handler.handleChirpAdd(ctx, ADD_CHIRP);
+
+    verify(socialMediaRepo).addChirp(LAO_ID, CHIRP);
+    verifyNoMoreInteractions(socialMediaRepo);
   }
 
   @Test
-  public void testHandleDeleteChirp() throws DataHandlingException, UnknownLaoException {
-    MessageGeneral message = new MessageGeneral(SENDER_KEY, ADD_CHIRP, gson);
-    messageHandler.handleMessage(messageSender, chirpChannel, message);
+  public void testHandleDeleteChirp() throws UnknownLaoException, InvalidMessageIdException {
+    // Act as if chirp exist
+    when(socialMediaRepo.deleteChirp(any(), any())).thenReturn(true);
+    HandlerContext ctx = new HandlerContext(CHIRP_ID, SENDER, CHIRP_CHANNEL, messageSender);
 
-    final DeleteChirp DELETE_CHIRP = new DeleteChirp(message.getMessageId(), DELETION_TIME);
+    handler.handleDeleteChirp(ctx, DELETE_CHIRP);
 
-    MessageGeneral message2 = new MessageGeneral(SENDER_KEY, DELETE_CHIRP, gson);
-    messageHandler.handleMessage(messageSender, chirpChannel, message2);
-
-    LaoView laoView = laoRepo.getLaoView(LAO_ID);
-    Lao updatedLao = laoView.createLaoCopy();
+    verify(socialMediaRepo).deleteChirp(LAO_ID, CHIRP_ID);
+    verifyNoMoreInteractions(socialMediaRepo);
+  }
 
-    Optional<Chirp> chirpOpt = updatedLao.getChirp(message.getMessageId());
-    assertTrue(chirpOpt.isPresent());
-    Chirp chirp = chirpOpt.get();
+  @Test
+  public void testUnorderedDeleteChirp() {
```

Comment:
I don't understand the title of the test. Does it mean that the chirp is not present ?

Reply:
I'll take a look at it and give a better name